infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -225,6 +238,34 @@ public static void main(String[] args) {
         corfuServerThread.start();
     }
 
+    /**
+     * Fetches the IP address given an interface name.
+     * Throws an unrecoverable error and aborts if the server is unable to find a valid address.
+     *
+     * @param interfaceName Network interface name.
+     * @return address for the interface name.
+     */
+    private static String getAddressFromInterfaceName(String interfaceName) {
+        try {
+            Enumeration<InetAddress> inetAddress = NetworkInterface.getByName(interfaceName)
+                    .getInetAddresses();
```

Comment:
why do you care about ipv4 and loopback devices? can't you just take the SocketAddress object and pass it to netty ?

Reply:
The interface has both inet (ipv4) and inet6 (v6) addresses. So I need the check for ipv4 addresses. I can remove the loopback check.

Reply:
why do you care about that? just let it fail on the bind ?

Reply:
Each interface has ipv4 and ipv6 addresses and we are interested to bind to the v4 address only.